tests/testsuite/check_cfg.rs

```diff
@@ -317,6 +317,174 @@ fn well_known_names_values_doctest() {
         .run();
 }
 
+#[cargo_test(nightly, reason = "warning currently only on nightly")]
+fn test_false_lib() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2018"
+
+                [lib]
+                test = false
+            "#,
+        )
+        .file("src/lib.rs", "#[cfg(test)] mod tests {}")
+        .build();
+
+    p.cargo("check -v")
+        .with_stderr_does_not_contain(x!("rustc" => "cfg" of "docsrs,test"))
+        .with_stderr_contains(x!("rustc" => "cfg" of "docsrs"))
+        .with_stderr_data(str![[r#"
+...
+[WARNING] unexpected `cfg` condition name: `test`
+ --> src/lib.rs:1:7
```

Comment:
We should elide the body of the message to reduce the risk of rustc changing things and being blocked on cargo